include/dma.h

```diff
@@ -82,24 +83,30 @@ class DmaChannel {
 
 class DmaController {
 private:
-	Bit8u ctrlnum;
+	uint8_t ctrlnum;
 	bool flipflop;
 	DmaChannel *DmaChannels[4];
 public:
 	IO_ReadHandleObject DMA_ReadHandler[0x12];
 	IO_WriteHandleObject DMA_WriteHandler[0x12];
-	DmaController(Bit8u num) {
-		flipflop = false;
-		ctrlnum = num;		/* first or second DMA controller */
-		for(Bit8u i=0;i<4;i++) {
-			DmaChannels[i] = new DmaChannel(i+ctrlnum*4,ctrlnum==1);
-		}
+
+	DmaController(uint8_t num)
+	        : ctrlnum(num), // first or second DMA controller
+	          flipflop(false)
+	{
+		for (uint8_t i = 0; i < 4; ++i)
+			DmaChannels[i] = new DmaChannel(i + num * 4, num == 1);
```

Comment:
Can the 4 become a #define?

Reply:
It would be named something like `DMA_CHANNELS_PER_CONTROLLER` and would be used only in these few lines… I implemented an alternative solution using new `ARRAY_LEN` macro.